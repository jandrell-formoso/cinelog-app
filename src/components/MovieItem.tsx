/**
 * @format
 */

import React from 'react'
import { Text } from 'react-native-paper';

const MovieItem = ({ title, rating }: any) => {
  return (
    <>
      <Text>{title}</Text>
    </>
  )
}

export default MovieItem
