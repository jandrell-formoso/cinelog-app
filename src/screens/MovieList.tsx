/**
 * @format
 */
import React, { useState, useEffect } from "react";
import { Surface, Card } from "react-native-paper";
import { FlatList, Text } from "react-native";
import MovieItem from "../components/MovieItem";
import { MovieApi } from "@cinelog/core";

const MovieList = (props: any) => {
  const [movies] = useMovieList();
  console.log("movies", movies)
  // deconstruct props
  const { data } = props;
  return (
    <Surface>
      <FlatList data={data} renderItem={() => <MovieItem />} />
    </Surface>
  );
};

const useMovieList = () => {
  const [movies, setMovies] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const result = await MovieApi.getMoviesByPopularity();
        setMovies(result);
        return result;
      } catch (error) {
        return error;
      }
    }
    fetchData();
  }, []);

  return [movies];
}

export default MovieList;
