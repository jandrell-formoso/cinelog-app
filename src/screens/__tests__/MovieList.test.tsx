import React from 'react'
import { FlatList } from 'react-native';
import MovieList from "../MovieList";
import MovieItem from "../../components/MovieItem";
import { shallow, mount } from 'enzyme';
import { Surface } from 'react-native-paper';

describe('Movie List', () => {
  const mockProps = {
    navigation: {
      navigate: jest.fn(),
    }
  }
  it("should render correctly", () => {
    // create a snapshot
    const component = shallow(<MovieList {...mockProps} />);
    expect(component).toMatchSnapshot();
    expect(component.exists()).toBeTruthy();
  });
  // TODO: test flatlist renderItem
  it("should render 5 items in flatlist", () => {
    const flatlistProps = {
      ...mockProps,
      data: [
        { title: "Avengers 1" },
        { title: "Avengers 2" },
        { title: "Avengers 3" },
        { title: "X-men 1" },
        { title: "X-men 2" },
      ],
    }
    const wrapper = shallow(<MovieList {...flatlistProps} />);
    expect(wrapper).toMatchSnapshot();
  });
})
