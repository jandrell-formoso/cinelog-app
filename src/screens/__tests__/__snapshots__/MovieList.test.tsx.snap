// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie List should render 5 items in flatlist 1`] = `
<withTheme(Surface)>
  <FlatList
    data={
      Array [
        Object {
          "title": "Avengers 1",
        },
        Object {
          "title": "Avengers 2",
        },
        Object {
          "title": "Avengers 3",
        },
        Object {
          "title": "X-men 1",
        },
        Object {
          "title": "X-men 2",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</withTheme(Surface)>
`;

exports[`Movie List should render correctly 1`] = `
<withTheme(Surface)>
  <FlatList
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</withTheme(Surface)>
`;
